.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,150 @@
+{
+  "name": "mne-dev",
+  // "image": "mcr.microsoft.com/devcontainers/base:debian-11",
+  "build": { "dockerfile": "Dockerfile" },
+  "containerEnv": {
+    "PYTHONNOUSERSITE": "true", // Make Python ignore the user's site-packages folder if it exists
+    "XDG_RUNTIME_DIR": "/home/vscode/.cache/xdgr"
+  },
+  "features": {
+    // See https://containers.dev/features
+    // APT packages
+    "ghcr.io/rocker-org/devcontainer-features/apt-packages:1": {
+      "packages": "mesa-utils,libegl1,^libxcb.*-dev,libx11-xcb-dev,libglu1-mesa-dev,libxrender-dev,libxi-dev,libxkbcommon-dev,libxkbcommon-x11-dev"
+    },
+    // Git
+    "ghcr.io/devcontainers/features/git:1": {
+      "version": "os-provided",
+      "ppa": "true"
+    },
+    // Conda and Mamba
+    "ghcr.io/mamba-org/devcontainer-features/micromamba:1": {
+      "channels": "conda-forge",
+      "packages": "conda mamba ipython jupyterlab mne"
+    },
+    // Docker interface with host's Docker service
+    "ghcr.io/devcontainers/features/docker-outside-of-docker:1": {},
+    // Zsh plugins
+    "ghcr.io/devcontainers-contrib/features/zsh-plugins:0": {
+      "plugins": "zsh-autosuggestions zsh-syntax-highlighting history-substring-search",
+      "omzPlugins": "https://github.com/zsh-users/zsh-autosuggestions https://github.com/zsh-users/zsh-syntax-highlighting"
```

Comment:
why are these needed?  Will the user ever access a terminal in the container (and if so, won't it be a BASH terminal by default?)

Reply:
Yes, of course :) It's a complete development environment, and if you open a terminal, it will open the "remote" terminal inside the Docker container.

The default shell is zsh with oh-my-zsh.

I'm trying to provide a nice and comfy shell experience through these plugins.

Reply:
> I'm trying to provide a nice and comfy shell experience through these plugins.

Another word for "comfy" is "familiar". I don't necessarily object to including `zsh` alongside `bash` but I'm dubious about making `zsh` the default here.

Reply:
MacOS users have `zsh` as the default shell and zsh is very close to bash.

Reply:
yes, I know that since 2019 macOS uses zsh as the default shell. And I've heard that they're similar. What I'm questioning is why tailor the experience *in a linux container* to feel familiar to macOS users.  Sticking with BASH seems like the natural default choice, and deviation from that feels like it needs to be justified.

Reply:
Both shells are first-class citizens in the MS dev containers. By default, both are shipped, hence I'm shipping both too. The plugins listed here make the zsh experience better and superior to the Bash experience. I have not yet decided which shell shall be the default one. However, I'm surprised this seems to be so controversial. VS Code clearly displays which shell is being used, and when spawning a new terminal, one can select the shell from a dropdown menu. Besides, I don't see how the choice of shell matters as long as one doesn't do shell programming; and this is an MNE dev container, no generic Linux Shell Programming dev container. Like I said, I haven't decided yet which default I think would be best; but It's quite obvious to me though that the zsh prompt looks better and the shell provides more help (history substring search being one of those features) during interactive use.

Reply:
> I have not yet decided which shell shall be the default one.

This is news to me; you originally set `zsh` as default (which I noticed at the time), then removed that setting in 7d5c6a2097d348460c79a95cb0a1b1a955ceeba8 *after* my earlier comments. It seems a bit disingenuous to now say "why is this so controversial, they're both included and I haven't chosen a default".

That said, I'm happy if both are included and it's easy (and documented) for users to switch the shell they get in their container. Ideally this would be a one-time config (not something that needs doing every time the container is launched) if that's possible

Reply:
> This is news to me; you originally set `zsh` as default (which I noticed at the time), then removed that setting in [7d5c6a2](https://github.com/mne-tools/mne-python/commit/7d5c6a2097d348460c79a95cb0a1b1a955ceeba8) _after_ my earlier comments

Ah, I see!! No, this was actually a misunderstanding. The default shell always was Bash, and the setting you're referring to controls which shell should be spawned by default when creating a new terminal without explicitly selecting a shell. This was in fact an accidental setting that I copied over from another dev container config I'm working with. But this setting never changed the (system) default shell to zsh, nor dropped bash or something :) I had however forgotten to properly set up `mamba` and `conda` for bash – something I realized and fixed after your comment / question.



> That said, I'm happy if both are included and it's easy (and documented) for users to switch the shell they get in their container. Ideally this would be a one-time config (not something that needs doing every time the container is launched) if that's possible

Yep, I'm open for this! for now I'm really still trying to make things work and experimenting a bit! Let's save the heated debates for later! 😁🤗

Oh and, happy holidays to you all!!!